ui/src/app/shared/services/workflows-service.ts

```diff
@@ -14,15 +14,15 @@ export class WorkflowsService {
             .then(res => res.body as Workflow);
     }
 
-    public list(phases: string[], namespace: string) {
-        return requests.get(`api/v1/workflows/${namespace}?${this.queryParams({phases}).join('&')}`).then(res => res.body as WorkflowList);
+    public list(namespace: string, phases: string[], labels: string[]) {
+        return requests.get(`api/v1/workflows/${namespace}?${this.queryParams({phases, labels}).join('&')}`).then(res => res.body as WorkflowList);
     }
 
     public get(namespace: string, name: string) {
         return requests.get(`api/v1/workflows/${namespace}/${name}`).then(res => res.body as Workflow);
     }
 
-    public watch(filter: {namespace?: string; name?: string; phases?: Array<string>}): Observable<models.kubernetes.WatchEvent<Workflow>> {
+    public watch(filter: {namespace?: string; name?: string; phases?: Array<string>; labels?: Array<string>}): Observable<models.kubernetes.WatchEvent<Workflow>> {
```

Comment:
You add `labels` as a parameter but don't actually change the request?

Reply:
`request` uses the whole `filter` as its parameter, which includes `labels`.

Reply:
It uses the whole `filter` as a parameter, no need to change the request.